components/system/ThemeProvider.js

```diff
@@ -0,0 +1,36 @@
+import * as React from "react";
+import * as Constants from "~/common/constants";
+
+import { ThemeProvider as EmotionTP } from "@emotion/react";
+
+export default function ThemeProvider({ children }) {
+  const [theme, setTheme] = React.useState({ darkmode: true });
+  const toggleDarkMode = (e) => setTheme((prev) => ({ ...prev, darkmode: e.detail.darkmode }));
+
+  React.useEffect(() => {
+    if (!window) return;
+    window.addEventListener("slate-theme-toggle-darkmode", toggleDarkMode);
```

Comment:
Can we make this more generic? `set-slate-theme` where it defaults to light.

Reply:
You got it, I made it more generic